lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -135,9 +136,19 @@ def _on_relation_changed(self, event) -> None:
         )
         if int_tls_ca:
             relation_data[INT_TLS_CA_KEY] = int_tls_ca
-
         self.database_provides.update_relation_data(event.relation.id, relation_data)
 
+    def _on_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handles providing mongos with KeyFile and hosts."""
+        # First we need to ensure that the database requested event has run
+        # otherwise we would write secrets in plain sight.
+        if not self.database_provides.fetch_relation_field(event.relation.id, "database"):
+            logger.info("Database Requested has not run yet, skipping.")
+            event.defer()
+            return
```

Comment:
The above should be removed as soon as moving on to the new version of the libs that I hope to merge next week :-)

This will be addressed on the lib level as highlighted on the [corresponding issue](https://github.com/canonical/data-platform-libs/issues/193)